pkg/cloudprovider/azure.go

```diff
@@ -562,26 +581,65 @@ func (a *Azure) getAuthorizer(env azureapi.Environment, clientID, clientSecret,
 		cred azcore.TokenCredential
 		err  error
 	)
-	if a.azureWorkloadIdentityEnabled && strings.TrimSpace(clientSecret) == "" {
-		options := azidentity.WorkloadIdentityCredentialOptions{
+
+	// MSI Override for ARO HCP
+	msi := os.Getenv("AZURE_MSI_AUTHENTICATION")
+	if msi == "true" {
+		options := azidentity.ManagedIdentityCredentialOptions{
 			ClientOptions: azcore.ClientOptions{
 				Cloud: cloudConfig,
 			},
-			ClientID:      clientID,
-			TenantID:      tenantID,
-			TokenFilePath: tokenFile,
 		}
-		cred, err = azidentity.NewWorkloadIdentityCredential(&options)
+
+		var err error
+		cred, err = azidentity.NewManagedIdentityCredential(&options)
```

Comment:
Can you explain why isn't the clientID set even when it is provided?

Reply:
Yeah, Microsoft is building us an additional sidecar container that will take the MSI authentication call before it goes to IDMS, modify it a bit, approve it, and send back a token. Basically it will add info for the HC ID, mgmt cluster info, and the MSI client ID. 

So the operator itself doesn't need to know anything, just making the IDMS call and the sidecar container handles the rest of it.